lib/internal/Magento/Framework/Stdlib/DateTime/Timezone.php

```diff
@@ -162,7 +162,7 @@ public function date($date = null, $locale = null, $useTimezone = true)
 
         if (empty($date)) {
             return new \DateTime('now', new \DateTimeZone($timezone));
-        } elseif ($date instanceof \DateTime) {
+        } elseif ($date instanceof \DateTimeInterface) {
             return $date->setTimezone(new \DateTimeZone($timezone));
```

Comment:
I think the only thing we can do here for best DX is to internally convert the `DateTimeImmutable` to `DateTime` and enforce the return type hint of `DateTime` always